--- build/configure	2016-01-28 12:45:32.444094587 -0500
+++ build/configure	2016-01-28 13:12:09.796051190 -0500
@@ -2969,7 +2969,6 @@
     as_fn_error $? "Your 'rm' program is bad, sorry." "$LINENO" 5
   fi
 fi
-
 ac_config_headers="$ac_config_headers config.h"
 
 
@@ -5123,12 +5122,6 @@
 then
     org_cflags=$CFLAGS
     CFLAGS=-m64
-    if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -5142,7 +5135,7 @@
 }
 
 _ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
+if ac_fn_c_try_compile "$LINENO"; then :
 
       CFLAGS="-m64 $org_cflags"
 
@@ -5155,8 +5148,6 @@
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
 
-fi
-
 # Issue 213: Search for clock_gettime to help people linking
 #            with a static version of libevent
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing clock_gettime" >&5
@@ -5901,7 +5892,16 @@
   $as_echo_n "(cached) " >&6
 else
 
-  if test "$cross_compiling" = yes; then :
+  if test $ac_compiler_gnu = yes; then
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking with GNU flags" >&5
+$as_echo_n "checking with GNU flags... " >&6; }
+    if $CC -E -dM - < /dev/null | grep LITTLE_ENDIAN > /dev/null; then
+      ac_cv_c_endian=little
+    else
+      ac_cv_c_endian=big
+    fi
+  else
+    if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -5924,17 +5924,18 @@
 _ACEOF
 if ac_fn_c_try_run "$LINENO"; then :
 
-    ac_cv_c_endian=big
+      ac_cv_c_endian=big
 
 else
 
-    ac_cv_c_endian=little
+      ac_cv_c_endian=little
 
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
 
+  fi
 
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_c_endian" >&5
