--- a/configure.ac.orig	2015-02-25 11:27:46.000000000 +0000
+++ b/configure.ac	2015-06-18 23:15:28.410567298 +0000
@@ -370,6 +370,9 @@
 	*-*-linux*)
 		ac_sys_system=Linux
 		;;
+	*-rumprun-netbsd*)
+		ac_sys_system=NetBSD
+		;;
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
@@ -415,6 +418,9 @@
 			_host_cpu=$host_cpu
 		esac
 		;;
+	*-*-netbsd*)
+		_host_cpu=$host_cpu
+		;;
 	*-*-cygwin*)
 		_host_cpu=
 		;;
