--- a/Modules/Setup.dist	2015-11-11 20:35:02.399927179 -0500
+++ b/Modules/Setup.dist	2015-11-11 20:35:21.183927097 -0500
@@ -391,3 +391,12 @@
 
 # Another example -- the 'xxsubtype' module shows C-level subtyping in action
 xxsubtype xxsubtype.c
+
+SQLITE_MODULE_NAME=='"sqlite3"'
+_sqlite3 \
+  _sqlite/cache.c _sqlite/microprotocols.c _sqlite/row.c \
+  _sqlite/connection.c  _sqlite/module.c            _sqlite/statement.c \
+  _sqlite/cursor.c      _sqlite/prepare_protocol.c  _sqlite/util.c \
+  -IModules/_sqlite \
+  -I../../sqlite -L../../pkgs/lib -lsqlite3 \
+  -DMODULE_NAME$(SQLITE_MODULE_NAME)
